AirCastingTests/Logic/SDSync/SdSyncAveragingServiceTests.swift

```diff
@@ -0,0 +1,135 @@
+            measurements.append(SDSyncMeasurement(measuredAt: startDate + TimeInterval(i), value: Double(i)))
+        }
+        
+        var resultMeasurements: [SDSyncMeasurement] = []
+        
+        let reminder = service.averageMeasurementsWithReminder(measurements: measurements, startTime: startDate, averagingWindow: thresholdWindow, action: { measurement, _ in resultMeasurements.append(measurement) })
+        
+        XCTAssertEqual(resultMeasurements.map(\.value), expectedAverages)
+    }
+    
+    func test_averageMeasurementsWithReminderCalledForSecondWindow_calculatesRightAverages() throws {
+        let thresholdWindow = AveragingWindow.secondThresholdWindow
+        let numberOfIntervals = 6
+        let startDate = Date.init(timeInterval: TimeInterval(-thresholdWindow.rawValue * numberOfIntervals), since: Date())
+        
+        var measurements: [SDSyncMeasurement] = []
+        var expectedAverages: [Double] = []
+        var valuesForAveraging: [Double] = []
+        
+        for i in 0...numberOfIntervals*thresholdWindow.rawValue {
+            if i != 0 && i%thresholdWindow.rawValue == 0 {
+                expectedAverages.append(valuesForAveraging.reduce(0.0, +) / Double(valuesForAveraging.count))
+                valuesForAveraging = []
+            }
+            valuesForAveraging.append(Double(i))
+            measurements.append(SDSyncMeasurement(measuredAt: startDate + TimeInterval(i), value: Double(i)))
+        }
+        
+        var resultMeasurements: [SDSyncMeasurement] = []
+        
+        _ = service.averageMeasurementsWithReminder(measurements: measurements, startTime: startDate, averagingWindow: thresholdWindow, action: { measurement, _ in resultMeasurements.append(measurement) })
+        
+        XCTAssertEqual(resultMeasurements.map(\.value), expectedAverages)
+    }
+    
+    func test_averageMeasurementsWithReminderCalledWithBigGapBetweenMeasurements_calculatesRightAverages() throws {
+        let thresholdWindow = AveragingWindow.secondThresholdWindow
+        let numberOfIntervals = 60
+        let startDate = Date.init(timeInterval: TimeInterval(-thresholdWindow.rawValue * numberOfIntervals), since: Date())
+        
+        var measurements: [SDSyncMeasurement] = []
+        var expectedAverages: [Double] = []
+        var valuesForAveraging: [Double] = []
+        
+        for i in 0...numberOfIntervals*thresholdWindow.rawValue {
+            if i != 0 && !valuesForAveraging.isEmpty && i%thresholdWindow.rawValue == 0 {
+                expectedAverages.append(valuesForAveraging.reduce(0.0, +) / Double(valuesForAveraging.count))
+                valuesForAveraging = []
+            }
+            if i > 70 && i < numberOfIntervals*thresholdWindow.rawValue-100 {
+                continue
+            }
+            valuesForAveraging.append(Double(i))
+            measurements.append(SDSyncMeasurement(measuredAt: startDate + TimeInterval(i), value: Double(i)))
+        }
```

Comment:
Maybe we could change this `i` to `newValue` of something that could indicate broader context?